dropshot/src/handler.rs

```diff
@@ -197,6 +201,91 @@ pub trait Extractor: Send + Sync + Sized {
     ) -> ExtractorMetadata;
 }
 
+/**
+ * `WebSocketExt` defines a function for upgrading a RequestContext to a websocket
+ * connection. This automatically handles checking the original request's headers
+ * to make sure it is a valid websocket connection.
+ */
+#[async_trait]
+pub trait WebSocketExt {
```

Comment:
Come to think of it though, is there any reason this needs to be a separate trait and not just have upgrade as an inherent method on RequestContext? It's already in the same crate and I don't think we'd be implementing it for any other type

Reply:
Seems fine to me just didn't know if you wanted it top level supported.